src/coremods/core_channel/core_channel.cpp

```diff
@@ -132,6 +132,7 @@ class CoreModChannel : public Module, public CheckExemption::EventListener
 		// Build a map of limits to their mode character.
 		insp::flat_map<int, std::string> limits;
 		const ModeParser::ListModeList& listmodes = ServerInstance->Modes->GetListModes();
+		limits.reserve(listmodes.size());
```

Comment:
This is probably going to reserve way too much space as its indexed by limit and most people will have half a dozen limits or so.

This is also a reasonably cold codepath so a extra few allocations won't hurt here.